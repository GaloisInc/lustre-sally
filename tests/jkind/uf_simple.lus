-- This file exercises simple functions
-- Expected output:
-- VALID PROPERTIES: [ok]
-- INVALID PROPERTIES: [cex] (k = 1)

function f(x : int) returns (y : int);

node main() returns (ok, cex: bool);
var
  count : int;
let
  count = 1 + (0 -> pre count);
  assert f(count + 1) > f(count);

  ok = not (count = 20 and f(20) = 20 and f(1) = 2);
  --%PROPERTY ok;

  cex = not (count = 20 and f(20) = 20 and f(1) = 1);
  --%PROPERTY cex;
tel;
