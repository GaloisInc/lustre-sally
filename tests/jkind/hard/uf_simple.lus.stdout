[[1;91mError[0m] Parse error at 8:6

